lib/t/matches.ex

```diff
@@ -600,25 +742,95 @@ defmodule T.Matches do
     end
   end
 
-  def match_check() do
-    MatchEvents
-    |> distinct([m], m.match_id)
-    |> join(:inner, [m], ma in Match, on: ma.id == m.match_id)
-    |> where([m], m.timestamp < fragment("now() - INTERVAL '48 hours'"))
-    |> select([m, ma], {m.match_id, ma.user_id_1})
-    |> T.Repo.all()
-    |> Enum.map(fn {match_id, user_id} ->
-      T.Matches.unmatch_match(user_id, match_id)
-    end)
+  def notify_match_expiration_reset(match_id, user_ids) do
+    message = {__MODULE__, :expiration_reset, match_id}
+
+    for uid <- user_ids do
+      broadcast_for_user(uid, message)
+    end
   end
 
+  # Expired Matches
+
   def match_timeslot_new_event(match_id, event) do
     if match_id != [] and match_id != nil and match_id != "" do
-      Repo.insert(%MatchEvents{
+      Repo.insert(%MatchEvent{
         timestamp: DateTime.truncate(DateTime.utc_now(), :second),
         match_id: match_id,
         event: event
       })
     end
   end
+
+  def match_expired_check() do
+    expired_matches_q()
+    |> join(:inner, [m], ma in Match, on: ma.id == m.match_id)
+    |> where([m], m.timestamp < fragment("now() - INTERVAL '48 hours'"))
+    |> select([m, ma], {ma.user_id_1, ma.user_id_2})
+    |> T.Repo.all()
+    |> Enum.map(fn {user_id_1, user_id_2} ->
+      expire_match(user_id_1, user_id_2)
+    end)
+  end
+
+  def match_soon_to_expire_check() do
+    expired_matches_q()
+    |> join(:inner, [m], ma in Match, on: ma.id == m.match_id)
+    |> where([m], m.timestamp < fragment("now() - INTERVAL '46:1 HOURS:MINUTES'"))
+    |> where([m], m.timestamp > fragment("now() - INTERVAL '46 hours'"))
```

Comment:
not sure whether such approach will work reliably given the pruner interval is also 1 minute

Reply:
the problem is that we need to send this push only once (so e.g. expanding to 46:2 may result in two pushes)